src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -729,13 +729,9 @@ public void add(final AbstractHistogram otherHistogram) throws ArrayIndexOutOfBo
      */
     public void subtract(final AbstractHistogram otherHistogram)
             throws ArrayIndexOutOfBoundsException, IllegalArgumentException {
-        long highestRecordableValue = valueFromIndex(countsArrayLength - 1);
-        if (highestRecordableValue < otherHistogram.getMaxValue()) {
-            if (!isAutoResize()) {
-                throw new ArrayIndexOutOfBoundsException(
-                        "The other histogram includes values that do not fit in this histogram's range.");
-            }
-            resize(otherHistogram.getMaxValue());
+        if (highestEquivalentValue(otherHistogram.getMaxValue()) > getMaxValue()) {
```

Comment:
I _think_ this is the right condition: if the high end of this histogram's bucket for the other's max value exceeds the high end of the max value's bucket